stock/app.py

```diff
@@ -18,23 +18,51 @@
 app = Flask("stock-service")
 
 # Connect to DB
-db: redis.Redis = redis.Redis(host=os.environ['REDIS_HOST'],
-                              port=int(os.environ['REDIS_PORT']),
-                              password=os.environ['REDIS_PASSWORD'],
-                              db=int(os.environ['REDIS_DB']))
+db_0: redis.Redis = redis.Redis(host=os.environ['REDIS_HOST_0'],
+                                port=int(os.environ['REDIS_PORT']),
+                                password=os.environ['REDIS_PASSWORD'],
+                                db=int(os.environ['REDIS_DB']))
+
+db_1: redis.Redis = redis.Redis(host=os.environ['REDIS_HOST_1'],
+                                port=int(os.environ['REDIS_PORT']),
+                                password=os.environ['REDIS_PASSWORD'],
+                                db=int(os.environ['REDIS_DB']))
+
+db_2: redis.Redis = redis.Redis(host=os.environ['REDIS_HOST_2'],
+                                port=int(os.environ['REDIS_PORT']),
+                                password=os.environ['REDIS_PASSWORD'],
+                                db=int(os.environ['REDIS_DB']))
+
+db_shards = [db_0, db_1, db_2]
+MODULO_HASH = len(db_shards)
 
 
 def close_db_connection():
     """
     Close the DB connection
     """
-    db.close()
+    for db in db_shards:
+        db.close()
 
 
 # Run close_db_connection function when service ends
 atexit.register(close_db_connection)
 
 
+# Retrieve DB from item_id
+def get_db(item_id: str):
```

Comment:
I have two questions about the assignment of ids to dbs. Since we use the ```RANDOM_SEED = 444```, is it guaranteed that it generates a uniform distribution so that one db is not overloaded compared to another? Also, in the order service we used a different RANDOM_SEED so I was wondering whether we should keep them the same?

Reply:
- Random seed value should not affect the distribution of values
- The tests with 10/100/1000/10000 items resulted in a uniform distribution over the 3 shards